backend/prompt_studio/prompt_studio_core_v2/views.py

```diff
@@ -452,10 +454,24 @@ def upload_for_ide(self, request: HttpRequest, pk: Any = None) -> Response:
         serializer = FileUploadIdeSerializer(data=request.data)
         serializer.is_valid(raise_exception=True)
         uploaded_files: Any = serializer.validated_data.get("file")
+        file_converter = get_plugin_class_by_name(
+            name="file_converter",
+            plugins=self.processor_plugins,
+        )
+
         documents = []
         for uploaded_file in uploaded_files:
             # Store file
             file_name = uploaded_file.name
+            file_data = uploaded_file
+            file_type = uploaded_file.content_type
+            # Convert non-PDF files
+            if file_converter and file_type != "application/pdf":
+                file_data_bytes = uploaded_file.read()
+                with io.BytesIO(file_data_bytes) as file_stream:
+                    file_data = file_converter.convert_to_pdf(file_stream, file_name)
+                file_name = f"{os.path.splitext(file_name)[0]}.pdf"
```

Comment:
@jagadeeswaran-zipstack all this should be within the plugin as well ideally, we can discuss on this

Reply:
@chandrasekharan-zipstack  Like we discussed these logic moved to plugin module.